### PR TITLE
add control over indexability

### PR DESCRIPTION
**Pull request by nichtsam**
We don't over duplicate content from different environment to all be indexed by search engines, \
this PR adds control over indexability.

For seo safety reason, I decided it's better to allow indexing by default.

`<meta name="robots" content="noindex, nofollow" />` prevents HTML from being indexed.
`X-Robots-Tag` prevents all resource from being indexed, this is mainly for non-HTML resources, such as PDF files, video files, or image files.

## Test Plan

I don't see a test that brings much confidence here.

## Checklist

- [ ] Tests updated
- [ ] Docs updated

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -194,6 +196,7 @@ function Document({
 				<Meta />
 				<meta charSet="utf-8" />
 				<meta name="viewport" content="width=device-width,initial-scale=1" />
+				{disallowIndexing && <meta name="robots" content="noindex, nofollow" />}
```
Comment (kentcdodds) [inline app/root.tsx:199]:
```suggestion
				{allowIndexing ? null : <meta name="robots" content="noindex, nofollow" />}
```

File: .env.example [hunk 2]
```diff
@@ -13,3 +13,7 @@ SENTRY_DSN="your-dsn"
 GITHUB_CLIENT_ID="MOCK_GITHUB_CLIENT_ID"
 GITHUB_CLIENT_SECRET="MOCK_GITHUB_CLIENT_SECRET"
 GITHUB_TOKEN="MOCK_GITHUB_TOKEN"
+
+# set this to true to prevent search engines from indexing the website
+# default to allow indexing for seo safety
+DISALLOW_INDEXING="false"
```
Comment (kentcdodds) [inline .env.example:19]: I generally prefer variables not use negation. So this should instead be "ALLOW_INDEXING" which defaults to `true`

File: app/root.tsx [hunk 3]
```diff
@@ -181,11 +181,13 @@ function Document({
 	nonce,
 	theme = 'light',
 	env = {},
+	disallowIndexing,
```
Comment (kentcdodds) [inline app/root.tsx:184]: This should be:

```suggestion
	allowIndexing = true,
```

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Thanks! This looks pretty good. Just a simple suggestion.

**Comment by nichtsam** [comment]
I use `process.env.ALLOW_INDEXING !== 'false'` to get the boolean, so only explicitly setting it to false will disallow the indexing.
And I realized `.default()` isn't really safe in this case, because it would declare type as `"true" | "false"` while it's actually `"true" | "false" | undefined`, and since it's default to true, it's important to be aware of the `undefined` case.

**Review by kentcdodds — APPROVED** [review]
Super. Thank you!

**Pull request merged**